apps/meteor/ee/server/local-services/voip-freeswitch/service.ts

```diff
@@ -1,14 +1,64 @@
-import { type IVoipFreeSwitchService, ServiceClassInternal } from '@rocket.chat/core-services';
-import type { FreeSwitchExtension } from '@rocket.chat/core-typings';
-import { getDomain, getUserPassword, getExtensionList, getExtensionDetails } from '@rocket.chat/freeswitch';
+import { type IVoipFreeSwitchService, ServiceClassInternal, ServiceStarter } from '@rocket.chat/core-services';
+import type {
+	DeepPartial,
+	IFreeSwitchEventCall,
+	IFreeSwitchEventCaller,
+	IFreeSwitchEvent,
+	FreeSwitchExtension,
+	IFreeSwitchCall,
+	IFreeSwitchCallEventType,
+	IFreeSwitchCallEvent,
+	AtLeast,
+} from '@rocket.chat/core-typings';
+import { isKnownFreeSwitchEventType } from '@rocket.chat/core-typings';
+import { getDomain, getUserPassword, getExtensionList, getExtensionDetails, listenToEvents } from '@rocket.chat/freeswitch';
+import type { InsertionModel } from '@rocket.chat/model-typings';
+import { FreeSwitchCall, FreeSwitchEvent, Users } from '@rocket.chat/models';
+import { objectMap, wrapExceptions } from '@rocket.chat/tools';
+import type { WithoutId } from 'mongodb';
+import { MongoError } from 'mongodb';
 
 import { settings } from '../../../../app/settings/server';
 
 export class VoipFreeSwitchService extends ServiceClassInternal implements IVoipFreeSwitchService {
 	protected name = 'voip-freeswitch';
 
+	private serviceStarter: ServiceStarter;
+
 	constructor() {
 		super();
+
+		this.serviceStarter = new ServiceStarter(() => Promise.resolve(this.startEvents()));
+	}
+
+	private listening = false;
+
+	public async started(): Promise<void> {
+		this.onEvent('watch.settings', async ({ setting }): Promise<void> => {
```

Comment:
`this.onEvent` should be in `constructor` otherwise it won't work as expected..

Reply:
For my curiosity, what's the difference?

Reply:
it has to do on how the `LocalBroker` reads the service instance to register all it's events and methods:

- the constructor is called when you calls `registerService`
- the `LocalBroker` calls `created`
- it gets all registered events
- it gets all methods
- and then it calls `started`

https://github.com/RocketChat/Rocket.Chat/blob/f90b1c99bdf4d89350a70c9ef4c2d12ed3784fd8/packages/core-services/src/LocalBroker.ts#L59-L83